metrics/metrics.go

```diff
@@ -106,11 +122,42 @@ func Start(metricsBind string, pprof http.Handler) {
 		fmt.Fprintf(w, "GC run complete")
 	})
 	s.Handler = mux
-	log.DefaultLogger.Warn("metrics", "listen finished", "err", s.ListenAndServe())
+	go func() {
```

Comment:
Why running a go routine here ?

Reply:
`s.Serve` blocks forever. This method returns the listener so the caller can report (or tests can use) the actual address that listening ended up happening on.
It also means the caller can close the listener to stop the metrics task that got started, rather than not having any way to stop the spawned server.